tools/cli/internal/changelog/outputfilter/squash.go

```diff
@@ -112,27 +112,40 @@ func newSquashHandlers() []handler {
 }
 
 // EntryMappings groups entries by ID and then by OperationID and returns a Map[changeCode, Map[operationId, List[oasdiffEntry]]]
-func newEntriesMapPerIDAndOperationID(entries []*OasDiffEntry) map[string]map[string][]*OasDiffEntry {
-	result := make(map[string]map[string][]*OasDiffEntry)
+func newEntriesMapPerIDAndOperationID(entries []*OasDiffEntry) (result, hiddenResult map[string]map[string][]*OasDiffEntry) {
+	result = make(map[string]map[string][]*OasDiffEntry)
+	hiddenResult = make(map[string]map[string][]*OasDiffEntry)
 
 	for _, entry := range entries {
 		code := entry.ID
 		operationID := entry.OperationID
+		hidden := entry.HideFromChangelog
 
 		// Ensure the code map exists
 		if _, exists := result[code]; !exists {
 			result[code] = make(map[string][]*OasDiffEntry)
 		}
 
+		if _, exists := hiddenResult[code]; !exists {
+			hiddenResult[code] = make(map[string][]*OasDiffEntry)
+		}
```

Comment:
Shouldn't you create a new entry in the map only if there is at least one entry hidden? if yes., you could come this inside the hidden if

Reply:
we could but the way its being done for the visible ones is like this, both end up adding the code into the map before, even if there are no entries. in that case I can do that for both, I just wasn't sure if this statement existed to cover some corner case I'm not aware of

Reply:
I'm making the change and not finding any failures so lets try that!